### PR TITLE
Fix: cleanup noxfile and requirements

### LINKED ISSUE #466: Infrastructure cleanup: remove requirements.txt in liue of optional.dependencies table in pyproject.toml
After learning more about [dependency declaration in modern packaging](https://www.pyopensci.org/python-package-guide/package-structure-code/declare-dependencies.html), I am realizing that we could be using the pyproject.toml to declare all dependencies (test, development, docs etc) rather than having a separate requirements file that houses everything.

Doing this would allow people to only install subsets of deps that they wish OR they could choose to install everything . This would cleanup our noxfile steps a bit as well. 

I think we could also consolidate the nox docs static build with the live build into a single function.

so i propose a PR that just is a smallish cleanup pr that would

- [ ] Add dependencies as groups in an optional.dependencies table in our pyproject.toml file
- [ ] cleanup the nox file to then install using that approach + pip 
- [ ] Merge the noxfile docs live and docs static build into one with a flag to choose static vs live build. 

this is just a small effort towards easier maintenance (fewer files, less noxfile code etc)

### PR DESCRIPTION
**Pull request by lwasser**
closes #466 

## Description

This pr does a few things
✅  It cleans up our noxfile as i had a build for running in conda and we are not published on conda. I found a few other quirks as i was looking at it that i cleaned up
✅ it relaxes pinning pydantic vs just requiring <2.0 which should help users w/ environment resolution issues 
✅ removes requirements.txt files in favor of using the more modern pyproject.toml. This means that `pip install -e.[docs]` can be used to install docs. or `pip install -e.[docs, tests]` can be used to install our package and docs/tests requirements. [See this page](https://www.pyopensci.org/python-package-guide/package-structure-code/declare-dependencies.html) for an overview of dependencies and pyproject.toml 
✅ i also moved our tests directory to the root. this just makes things easier to find AND also further ensures we're testing on the installed version. 

we could create a large "install all requirements" group in the pyproject.toml as well if that is of interest

## Type of change

Select the statement best describes this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This is a documentation update
- [x] This is a infrastructure update (docs, ci, etc)
- [ ] Other (please describe)

## Does your PR include tests

If you are fixing a bug or adding a feature, we appreciate (but do not require)
tests to support whatever fix of feature you're implementing.

- [ ] Yes
- [ ] No, i'd like some help with tests
- [x] This change doesn't require tests

## Did you include your contribution to the change log?

- [x] Yes, `changelog.md` is up-to-date.

### CODE REVIEW

File: tests/integration/test_client.py [hunk 1]
```diff
@@ -5,15 +5,15 @@
 
 import pytest
 import responses
-from responses import matchers
-
 import stravalib.unithelper as uh
+from responses import matchers
 from stravalib.client import ActivityUploader
 from stravalib.exc import AccessUnauthorized, ActivityPhotoUploadFailed
```
Comment (lwasser) [inline tests/integration/test_client.py:11]:
```suggestion
from responses import matchers
import stravalib.unithelper as uh
from stravalib.client import ActivityUploader
from stravalib.exc import AccessUnauthorized, ActivityPhotoUploadFailed
```

**Comment by lwasser** [comment]
eeks. this looks bigger than it is. if y'all want me to break it into 2 prs
1. where i only remove requirements.txt and 
2. where i only move tests to the root dir 
please say the word

**Comment by lwasser** [comment]
something weird happened with code coverage. i get accurate reports in the terminal with pytest but not while running nox so looking into this

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -45,17 +45,17 @@ jobs:
...
+        # Need tags for setuptools_scm to provide a proper version
       - name: Fetch git tags
         run: git fetch origin 'refs/tags/*:refs/tags/*'
         # For now i made a separate tests run so we can see each version
         # build via github actions. There may be a better way to do this.
-      - name: Run tests with pytest & nox
+      - name: Run tests with pytest
         run: |
-          nox -s tests-${{ matrix.python-version }}
+          pytest --cov=src/stravalib/ --cov-report=xml:coverage.xml --cov-report=term tests/
```
Comment (lwasser) [inline .github/workflows/build-test.yml:58]: for some reason the coverage report is FINE when i run it outside of nox. let's see if that is the same in CI.

**Comment by lwasser** [comment]
🤷‍♀️ i'm giving up on fighting with nox. things work great with pytest. nox runs tests as expected but the code coverage element is reporting 0% coverage now and i'm not sure why. so for now i'm migrating our CI build to pytest. And i have an open question to the nox crew about why this might be happening. this is ready for review. when we merge i'll open a new issue about nox not working as expected.

**Comment by lwasser** [comment]
ok y'all i have scaled back this pr. It now ONLY updates the requirements and uses the pyproject.toml so we don't have to maintain additional requirement files. this should be easier to review. i'm going to submit another pr that updates the action versions (for security reasons). 

tests are building with text coverage again using nox. ready for review.

**Review by jsamoocha — APPROVED** [review]
Thanks, this looks good!

**Comment by lwasser** [comment]
awesome i'll merge @jsamoocha !!

**Pull request merged**